addons/context_actions/CfgContext.hpp

```diff
@@ -244,6 +244,34 @@ class EGVAR(context_menu,actions) {
             statement = QUOTE(_hoveredEntity setUnitLoadout configOf _hoveredEntity);
             icon = "\a3\3den\Data\Displays\Display3DEN\ToolBar\undo_ca.paa";
         };
+        class SwitchWeapon {
+            displayName = "$STR_A3_Switch1";
+            condition = QUOTE(2 <= {_x != ''} count [ARR_3(primaryWeapon _hoveredEntity, handgunWeapon _hoveredEntity, binocular _hoveredEntity)]);
+            statement = "";
+            icon = "\a3\ui_f\data\IGUI\Cfg\Actions\reammo_ca.paa";
+            class Rifle {
+                displayName = "$STR_A3_RSCDisplayArsenal_Tab_PrimaryWeapon";
+                condition = QUOTE(!isPlayer _hoveredEntity && {primaryWeapon _hoveredEntity != '' && {primaryWeapon _hoveredEntity != currentWeapon _hoveredEntity}});
```

Comment:
I'm thinking maybe all of these conditions should be extracted to separate functions?

Reply:
This part belongs to #520, sorry!

Reply:
978a9c1